ui/components/app/assets/asset-list/cells/asset-cell-badge.tsx

```diff
@@ -14,26 +14,32 @@ import {
   getMultichainIsEvm,
 } from '../../../../../selectors/multichain';
 import { getNetworkConfigurationsByChainId } from '../../../../../../shared/modules/selectors/networks';
-import { TokenFiatDisplayInfo } from '../../types';
 
-type TokenCellBadgeProps = {
-  token: TokenFiatDisplayInfo;
+type AssetCellBadgeProps = {
+  chainId: `0x${string}` | `${string}:${string}`;
+  isNative?: boolean;
+  tokenImage: string;
+  symbol: string;
 };
 
-export const TokenCellBadge = React.memo(
-  ({ token }: TokenCellBadgeProps) => {
+export const AssetCellBadge = React.memo(
+  ({ chainId, isNative, tokenImage, symbol }: AssetCellBadgeProps) => {
     const isEvm = useSelector(getMultichainIsEvm);
     const allNetworks = useSelector(getNetworkConfigurationsByChainId);
 
+    const avatarTokenSrc =
+      isEvm && isNative ? getNativeCurrencyForChain(chainId) : tokenImage;
+    // TODO: Fix in https://github.com/MetaMask/metamask-extension/issues/31880
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    const badgeWrapperSrc = getImageForChainId(chainId) || undefined;
```

Comment:
nit (we can leave it) - since this was moved, we could potentially fix?
```suggestion
    const badgeWrapperSrc = getImageForChainId(chainId) ?? undefined;
```